core/src/main/java/com/google/errorprone/bugpatterns/ReturnValueIgnored.java

```diff
@@ -204,8 +204,16 @@ private static boolean functionalMethod(ExpressionTree tree, VisitorState state)
   private static final Matcher<ExpressionTree> OPTIONAL_METHODS =
       anyOf(
           staticMethod().onClass("java.util.Optional"),
+          instanceMethod().onExactClass("java.util.Optional").named("filter"),
+          instanceMethod().onExactClass("java.util.Optional").named("flatMap"),
+          instanceMethod().onExactClass("java.util.Optional").named("get"),
+          instanceMethod().onExactClass("java.util.Optional").named("isEmpty"),
           instanceMethod().onExactClass("java.util.Optional").named("isPresent"),
-          instanceMethod().onExactClass("java.util.Optional").named("isEmpty"));
+          instanceMethod().onExactClass("java.util.Optional").named("map"),
+          instanceMethod().onExactClass("java.util.Optional").named("or"),
+          instanceMethod().onExactClass("java.util.Optional").named("orElse"),
+          instanceMethod().onExactClass("java.util.Optional").named("orElseGet"),
+          instanceMethod().onExactClass("java.util.Optional").named("orElseThrow"));
```

Comment:
Calling `orElseThrow` to assert an `Optional` is present and then discarding the return result appears to be more common than the other examples here. I'm not really defending that as a pattern, just saying that making this one an error maybe not be lower priority for us. I'm curious if you've run into examples of that?

Reply:
Occasionally, but rarely.

Reply:
As a reference, SonarLint requires that return values are used for all `Optional` methods that return non-void values.
https://github.com/SonarSource/sonar-java/blob/d59d157c852ac924f3029ddb8c0aa366501f9fda/java-checks/src/main/java/org/sonar/java/checks/IgnoredReturnValueCheck.java#L68